idseq_dag/engine/pipeline_step.py

```diff
@@ -49,6 +50,18 @@ def run(self):
     def count_reads(self):
         ''' count reads '''
 
+    def count_reads_work(self):
+        '''
+        Count reads for outputs consisting of sequence records.
+
+        Assumptions:
+        - For paired inputs, the outputs to be counted are the first two in the target.
+        - When the pipeline input is single-ended, the target consists of only a single
+          output, which is to be counted.
+        '''
+        files_to_count = self.output_files_local()[0:2] # will be either 1 or 2 files
```

Comment:
hmm since the actual code is only two lines, I don't think we need this function at all.  (mapping the output files here is also dangerous.) You can move the comments and documentation to count_reads()